sources/packages/api/src/route-controllers/user/user.controller.ts

```diff
@@ -1,20 +1,49 @@
-import { Controller, Get, Req } from "@nestjs/common";
+import { Controller, Get, Patch, Req } from "@nestjs/common";
 import { Request } from "express";
 import { AuthService, UserService } from "../../services";
+import BaseController from "../BaseController";
 
 @Controller("users")
-export class UserController {
+export class UserController extends BaseController {
   constructor(
     private readonly service: UserService,
     private readonly authService: AuthService,
-  ) {}
+  ) {
+    super();
+  }
 
   @Get("/check-user")
-  checkUser(@Req() request: Request) {
-    const userInfo = this.authService.parseAuthorizationHeader(
-      request.headers.authorization,
-    );
+  async checkUser(@Req() request: Request) {
+    try {
+      const userInfo = this.authService.parseAuthorizationHeader(
+        request.headers.authorization,
+      );
+      const userInSABC = await this.service.getUser(userInfo.userName);
+      if (!userInSABC) {
+        //TODO: Set a 404 in response when no user is found.
+        //response.status(404);
+      } else {
+        return userInSABC;
+      }
+    } catch (error) {
```

Comment:
Can we check if there is a global error handler or create a tech debt for it. In general, I would avoid adding a try/catch block in every single controller method.

Reply:
Created a Tech Debt:157
https://app.zenhub.com/workspaces/student-information-management-system-5fce9df5aa1b45000e937014/issues/bcgov/sims/157